vault/request_handling.go

```diff
@@ -578,6 +578,35 @@ func (c *Core) handleRequest(ctx context.Context, req *logical.Request) (retResp
 		return
 	}
 
+	// If this is a wrapping request that contains a wrapping token, check for
+	// token validity. We perform this before c.checkToken since wrapping token
+	// validation performs a token store lookup, and can potentially modify
+	// logical.Request's token-related fields.
+	switch req.Path {
+	case "sys/wrapping/rewrap", "sys/wrapping/unwrap":
+		valid, err := c.validateWrappingToken(ctx, req)
+
+		// Perform audit logging before returning if there's an issue with checking
+		// the wrapping token
+		if err != nil || !valid {
+			logInput := &audit.LogInput{
+				Auth:               req.Auth,
```

Comment:
One thing that I'm not sure of is what to pass into `audit.LogInput.Auth`. `req.Auth` is technically still nil at this point. We could either leave that out, or copy over the req.ClientToken, req.ClientAccessor, etc. over locally and output that under the auth block.